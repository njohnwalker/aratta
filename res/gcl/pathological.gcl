@req(x >= 0)
res := 0;
x0 := 0;
if x == 0 -> if fi
[] x == 1 -> res := y;
[] x >  1 ->
    do @inv()
    [] ~ x == x0 -> res, x0 := res + y, x0 - 1;
    od
fi
res := res / y;
@ens(res == x)