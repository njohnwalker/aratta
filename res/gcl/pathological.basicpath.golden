Path 1
  [ Branch 1
      [ 
      ]
      { HOLE }
    Branch 2
      ; Assume x >= 0
      ; Substitute
          @res -> 0
      ; Substitute
          @x0 -> 0
      ; Assume x == 1
      ; Substitute
          @res -> y
      { HOLE }
    Branch 3
      ; Assume x >= 0
      ; Substitute
          @res -> 0
      ; Substitute
          @x0 -> 0
      ; Assume x > 1
      { Satisfies INVARIANT == PLACEHOLDER }
  ]
  ; Substitute
      @res -> res / y
  { Satisfies res == x }
Path 2
  [ Branch 1
      [ 
      ]
      { HOLE }
    Branch 2
      ; Assume x >= 0
      ; Substitute
          @res -> 0
      ; Substitute
          @x0 -> 0
      ; Assume x == 1
      ; Substitute
          @res -> y
      { HOLE }
    Branch 3
      ; Assume INVARIANT == PLACEHOLDER
      [ Branch 1
          ; Assume ~ x == x0
          ; Substitute
              @res -> res + y
              @x0 -> x0 - 1
          { HOLE }
      ]
      { Satisfies INVARIANT == PLACEHOLDER }
  ]
  ; Substitute
      @res -> res / y
  { Satisfies res == x }
Path 3
  [ Branch 1
      [ 
      ]
      { HOLE }
    Branch 2
      ; Assume x >= 0
      ; Substitute
          @res -> 0
      ; Substitute
          @x0 -> 0
      ; Assume x == 1
      ; Substitute
          @res -> y
      { HOLE }
    Branch 3
      ; Assume INVARIANT == PLACEHOLDER
      ; Assume ~ ~ x == x0
      { HOLE }
  ]
  ; Substitute
      @res -> res / y
  { Satisfies res == x }